### PR TITLE
Update parser, hover improvements

### PR DESCRIPTION
**Pull request by JordanBoltonMN**
* Updated to a new version for powerquery-parser. Changes still need to be added to make use of the external type resolver callback.
* Mostly improvements on hover, specifically returning no hover on identifiers that are no values.

### CODE REVIEW

File: src/powerquery-language-services/analysis/analysis.ts [hunk 1]
```diff
@@ -6,7 +6,7 @@ import type { CompletionItem, Hover, SignatureHelp } from "vscode-languageserver
 import { IDisposable } from "../commonTypes";
 
 export interface Analysis extends IDisposable {
-    getCompletionItems(): Promise<ReadonlyArray<CompletionItem>>;
+    getCompletionItems(): Promise<CompletionItem[]>;
```
Comment (mattmasson) [inline src/powerquery-language-services/analysis/analysis.ts:9]: We seem to be inconsistent on the use of `[]` vs. `ReadonlyArray`. I prefer `[]` for the sake of brevity, but I'm not exactly sure what the difference ends up being.

File: src/powerquery-language-services/analysis/analysisBase.ts [hunk 2]
```diff
@@ -163,6 +167,30 @@ export abstract class AnalysisBase implements Analysis {
         );
     }
 
+    private static isValidHoverIdentifier(activeNode: PQP.Inspection.ActiveNode): boolean {
+        const ancestry: ReadonlyArray<PQP.Parser.TXorNode> = activeNode.ancestry;
+        if (ancestry.length <= 1) {
+            return true;
+        }
+
+        const leaf: PQP.Parser.TXorNode = PQP.Assert.asDefined(ancestry[0]);
```
Comment (mattmasson) [inline src/powerquery-language-services/analysis/analysisBase.ts:176]: Remind me what a `GeneralizedIdentifier` is again?

**Review by mattmasson — APPROVED** [review]

File: src/powerquery-language-services/analysis/analysis.ts (hunk 1 above)
Comment (JordanBoltonMN) [reply src/powerquery-language-services/analysis/analysis.ts:9]: In short, use`Readonly<T>` internally, and use `[]` for VSCode APIs.

You've probably noticed, but in general I prefer opt-in mutability. A bit ago I refactored the language-services layer to use `ReadonlyArray<T>` everywhere. I'm pulling back a bit on that so wherever we interface with VSCode APIs it returns `[]` as that's what the API expects.

File: src/powerquery-language-services/analysis/analysisBase.ts (hunk 2 above)
Comment (JordanBoltonMN) [reply src/powerquery-language-services/analysis/analysisBase.ts:176]: Generalized identifiers are identifiers that can have spaces between them and don't restrict keyword usage. Most commonly they are field names when dealing with records/tables, eg.

`table[a multi word identifier]`
`table[date]`

**Pull request merged**